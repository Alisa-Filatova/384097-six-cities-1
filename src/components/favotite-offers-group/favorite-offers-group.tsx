import * as React from 'react';
import OfferCard from '../offer-card/offer-card';
import CityTab from '../city-tab/city-tab';
import {CityOffersGroup, Offer} from '../../types/offer';

interface Props {
  offersGroup: CityOffersGroup;
  onFavoriteClick: (offer: Offer) => void;
  isAuthenticated: boolean;
}

const FavoriteOffersGroup = (props: Props) => {
  const {offersGroup, onFavoriteClick, isAuthenticated} = props;

  return (
    <div className="favorites__list">
      {Object.entries(offersGroup).map(([key, value]) => (
        <div
          className="favorites__locations-items"
          key={key}
        >
          <div className="favorites__locations locations locations--current">
            <CityTab city={key} isActive />
          </div>
          <div className="favorites__places">
            {Object.values(value).map((offer, idx) => (
              <OfferCard
                offer={offer}
                onFavoriteClick={onFavoriteClick}
                isAuthenticated={isAuthenticated}
                key={`${offer.title}${idx}`}
                prefix="favorites"
                small
              />
            ))}
          </div>
        </div>
      ))}
    </div>
  );
};

export default FavoriteOffersGroup;
