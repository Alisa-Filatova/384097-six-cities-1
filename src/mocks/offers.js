export default [
  {
    id: 123,
    title: `Beautiful & luxurious apartment at great location`,
    img: `img/apartment-01.jpg`,
    isPremium: true,
    price: 120,
    stars: 4,
    type: `Apartment`,
    isInBookmarks: false,
  },
  {
    id: 234,
    title: `Wood and stone place`,
    img: `img/apartment-02.jpg`,
    isPremium: false,
    price: 80,
    stars: 2,
    type: `Private room`,
    isInBookmarks: true,
  },
  {
    id: 456,
    title: `Canal View Prinsengracht`,
    img: `img/apartment-03.jpg`,
    isPremium: false,
    price: 180,
    stars: 3,
    type: `Apartment`,
    isInBookmarks: false,
  },
  {
    id: 567,
    title: `Nice, cozy, warm big bed apartment`,
    img: `img/apartment-01.jpg`,
    isPremium: false,
    price: 100,
    stars: 4,
    type: `Private room`,
    isInBookmarks: true,
  },
];
