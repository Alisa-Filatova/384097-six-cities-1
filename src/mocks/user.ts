export const userMock = {
  avatarUrl: `/static/avatar/4.jpg`,
  email: `alicefill@y888.ru`,
  id: 1,
  isPro: false,
  name: `alicefill`,
};
