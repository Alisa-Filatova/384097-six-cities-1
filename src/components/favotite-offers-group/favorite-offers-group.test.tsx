import * as React from 'react';
import {MemoryRouter as Router} from 'react-router-dom';
import * as renderer from 'react-test-renderer';
import FavoriteOffersGroup from './favorite-offers-group';
import {favoriteOffersGroupMock} from '../../mocks/favorite-offers-group';

describe(`FavoriteOffersGroup`, () => {
  it(`renders correctly`, () => {
    const group = renderer.create(
        <Router>
          <FavoriteOffersGroup
            offersGroup={favoriteOffersGroupMock as any}
            onFavoriteClick={jest.fn()}
            isAuthenticated={true}
          />
        </Router>
    ).toJSON();

    expect(group).toMatchSnapshot();
  });
});
