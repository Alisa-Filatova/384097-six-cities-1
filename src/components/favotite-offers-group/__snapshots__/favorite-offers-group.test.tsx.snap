// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoriteOffersGroup renders correctly 1`] = `
<div
  className="favorites__list"
>
  <div
    className="favorites__locations-items"
  >
    <div
      className="favorites__locations locations locations--current"
    >
      <div
        className="locations__item"
      >
        <a
          className="locations__item-link tabs__item tabs__item--active"
          href="#"
          onClick={[Function]}
          style={
            Object {
              "pointerEvents": "none",
            }
          }
        >
          <span>
            Amsterdam
          </span>
        </a>
      </div>
    </div>
    <div
      className="favorites__places"
    >
      <article
        className="favorites__place-card favorites__card place-card"
      >
        <div
          className="favorites__image-wrapper place-card__image-wrapper"
        >
          <a
            onClick={[Function]}
            style={
              Object {
                "cursor": "default",
              }
            }
          >
            <img
              alt="Beautiful & luxurious studio at great location"
              className="place-card__image"
              height={110}
              src="img/1.png"
              width={150}
            />
          </a>
        </div>
        <div
          className="favorites__card-info place-card__info"
        >
          <div
            className="place-card__price-wrapper"
          >
            <div
              className="place-card__price"
            >
              <b
                className="place-card__price-value"
              >
                €
                120
              </b>
              <span
                className="place-card__price-text"
              >
                / night
              </span>
            </div>
            <button
              className="place-card__bookmark-button button place-card__bookmark-button--active"
              onClick={[Function]}
              type="button"
            >
              <svg
                className="place-card__bookmark-icon"
                height={19}
                width={18}
              >
                <use
                  xlinkHref="#icon-bookmark"
                />
              </svg>
              <span
                className="visually-hidden"
              >
                To bookmarks
              </span>
            </button>
          </div>
          <div
            className="place-card__rating rating"
          >
            <div
              className="place-card__stars rating__stars"
            >
              <span
                style={
                  Object {
                    "width": "100%",
                  }
                }
              />
              <span
                className="visually-hidden"
              >
                Rating
              </span>
            </div>
          </div>
          <h2
            className="place-card__name"
          >
            <a
              href="/offer/1"
              onClick={[Function]}
            >
              Beautiful & luxurious studio at great location
            </a>
          </h2>
          <p
            className="place-card__type"
          >
            apartment
          </p>
        </div>
      </article>
    </div>
  </div>
</div>
`;
